### DISCUSSION TITLE
Binding to indexer

### DISCUSSION THREAD

**Discussion by asmirnov82** [AvaloniaUI/Avalonia#17832] [Q&A] [answered]
Hello,

I would like to create a view model, that implements 'string this[string key] { get; set; }' indexing or provides property `Values` that implements it. In my view I use binding to this indexer:
```xaml
<TextBox	Text="{Binding [Text1]}"/>
``` 
or
```xaml
<TextBox	Text="{Binding Values[Text1]}"/>
``` 
That works fine and correct values are shown.

But now I need to change the value for [Text1] or Values[Text1] and trigger NotifyPropertyChanged event by calling `PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(property));`.

What name should I use as `property` to make it work? I tried different onces without any success

**Comment by msneijders** [comment, marked as answer]
This could work:

```csharp
class ViewModel: INotifyPropertyChanged
{
    Dictionary<string, string> _items = new Dictionary<string, string>()
    {
        { "key1", "value1" },
        { "key2", "value2" },
        { "key3", "value3" },
    };

    public string this[string key]
    {
        get
        {
            return _items[key];
        }

        set
        {
            _items[key] = value;
            base.OnPropertyChanged("Item");
        }
    }
}
```

In xaml:

```xaml
<TextBox Text="{Binding [key1]}" />
```

So using `Item` as property name.

In avalonia code in `AvaloniaDictionary`, I also see the use of `"Item[key1]"`, but I could not get it to work using that.

**Reply by asmirnov82** [reply]
@msneijders, thank you for such quick answer. This works!
